kubernetes/cloud-otelcol/config/config.alloy

```diff
@@ -0,0 +1,199 @@
+    	"delete_key(attributes, \"process.executable.path\")",
+    	"delete_key(attributes, \"process.pid\")",
+    	"delete_key(attributes, \"process.runtime.description\")",
+    	"delete_key(attributes, \"process.runtime.name\")",
+    	"delete_key(attributes, \"process.runtime.version\")",
+    ]
+  }
+    
+  metric_statements {
+    context    = "resource"
+    statements = [
+    	"delete_key(attributes, \"k8s.pod.start_time\")",
+    	"delete_key(attributes, \"os.description\")",
+    	"delete_key(attributes, \"os.type\")",
+    	"delete_key(attributes, \"process.command_args\")",
+    	"delete_key(attributes, \"process.executable.path\")",
+    	"delete_key(attributes, \"process.pid\")",
+    	"delete_key(attributes, \"process.runtime.description\")",
+    	"delete_key(attributes, \"process.runtime.name\")",
+    	"delete_key(attributes, \"process.runtime.version\")",
+    ]
+  }
+    
+  log_statements {
+    context    = "resource"
+    statements = [
+    	"delete_key(attributes, \"k8s.pod.start_time\")",
+    	"delete_key(attributes, \"os.description\")",
+    	"delete_key(attributes, \"os.type\")",
+    	"delete_key(attributes, \"process.command_args\")",
+    	"delete_key(attributes, \"process.executable.path\")",
+    	"delete_key(attributes, \"process.pid\")",
+    	"delete_key(attributes, \"process.runtime.description\")",
+    	"delete_key(attributes, \"process.runtime.name\")",
+    	"delete_key(attributes, \"process.runtime.version\")",
+    ]
+  }
+    
+  output {
+    metrics = [otelcol.processor.transform.add_resource_attributes_as_metric_attributes.input]
+    logs    = [otelcol.processor.batch.default.input]
+    traces  = [
+    	otelcol.processor.batch.default.input,
+    	otelcol.connector.host_info.default.input,
+    ]
+  }
+}
+    
+otelcol.connector.host_info "default" {
+  // https://grafana.com/docs/alloy/latest/reference/components/otelcol.connector.host_info/
+  host_identifiers = ["host.name"]
+    
+  output {
+    metrics = [otelcol.processor.batch.default.input]
+  }
+}
+    
+otelcol.processor.transform "add_resource_attributes_as_metric_attributes" {
+  // https://grafana.com/docs/alloy/latest/reference/components/otelcol.processor.transform/
+  error_mode = "ignore"
+    
+  metric_statements {
+    context    = "datapoint"
+    statements = [
+    	"set(attributes[\"deployment.environment\"], resource.attributes[\"deployment.environment\"])",
+    	"set(attributes[\"service.version\"], resource.attributes[\"service.version\"])",
+    ]
+  }
+    
+  output {
+    metrics = [otelcol.processor.batch.default.input]
+  }
+}
+    
+otelcol.processor.batch "default" {
+  // https://grafana.com/docs/alloy/latest/reference/components/otelcol.processor.batch/
+  output {
+    metrics = [otelcol.exporter.otlphttp.grafana_cloud.input]
+    logs    = [otelcol.exporter.otlphttp.grafana_cloud.input]
+    
+    // NOTE: Updated this from the initial implementation to send traces correctly
```

Comment:
@theSuess, this had to be changed to send traces.

Reply:
Unfortunately, no. I worked on the `cloud-dev` environment, which has a different setup.

Reply:
@theSuess I've added a Docker Compose setup deploying QP in a microservice architecture, similar to the Kubernetes setup.

Profiling works in the monolith Docker Compose setup, but it's also failing in the microservices version.

Reply:
The issue was the wrong config for `otelcol.auth.basic` and `otelcol.exporter.otlphttp` which is mentioned in the other thread. Resolving.